plugins/lsp/language-servers/default.nix

```diff
@@ -691,6 +691,32 @@ let
         "nodePackages"
         "@volar/vue-language-server"
       ];
+      extraOptions = {
+        tslsIntegration = mkOption {
+          type = types.bool;
+          description = ''
+            Enable integration with TypeScript language server.
+          '';
+          default = true;
+          example = false;
+        };
+      };
+      extraConfig = cfg: {
+        plugins.lsp.servers.ts-ls = {
+          filetypes = [ "vue" ];
```

Comment:
If we don't have a matching default for ts files this will disable them won't it?

Reply:
This block was intended to just append to the user's existing ts-ls config when the integration is enabled.

Reply:
Added a couple tests to show what I meant.

Reply:
The issue is that `ts-ls.filetypes` is `null` by default. This means that enabling the `tslsIntegration` but leaving the `ts-ls.filetypes` as default will result on only  `{ "vue" }` being passed to nvim-lspconfig. I think we need to set the propagate the default value of  `{ "javascript", "javascriptreact", "javascript.jsx", "typescript", "typescriptreact", "typescript.tsx" }`

Reply:
Oh, you mean the plugin's default has those values and we override the default plugin behavior by enabling it.

Reply:
Exactly, I think the fix is to hardcode the plugin's default as the nix default

Reply:
Sounds good, I'll update that when I get home later.